pkg/controllers/cisnetworkpolicy/controller.go

```diff
@@ -0,0 +1,166 @@
+package cisnetworkpolicy
+
+import (
+	"context"
+	"net"
+	"sort"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/rancher/k3s/pkg/server"
+	coreclient "github.com/rancher/wrangler-api/pkg/generated/controllers/core/v1"
+	core "k8s.io/api/core/v1"
+	netv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	FlannelPresenceLabel         = "flannel.alpha.coreos.com/public-ip"
+	FlannelHostNetworkPolicyName = "rke2-flannel-host-networking"
+)
+
+func CISNetworkPolicyController(ctx context.Context, sc *server.Context) error {
+	Register(ctx, sc.Core.Core().V1().Node(), sc.K8s)
+
+	return nil
+}
+
+func Register(ctx context.Context,
+	nodes coreclient.NodeController,
+	k8s kubernetes.Interface,
+) error {
+	h := &handler{
+		ctx: ctx,
+		k8s: k8s,
+	}
+	logrus.Debugf("CISNetworkPolicyController: Registering controller hooks")
+	nodes.OnChange(ctx, "cisnetworkpolicy-node", h.onChange)
+	nodes.OnRemove(ctx, "cisnetworkpolicy-node", h.onRemove)
+	return nil
+}
+
+type handler struct {
+	ctx context.Context
+	k8s kubernetes.Interface
+}
+
+func (h *handler) onRemove(key string, node *core.Node) (*core.Node, error) {
+	return h.reconcileFlannelHostNetworkPolicy(key, node)
```

Comment:
```suggestion
	if node == nil {
		return nil, nil
	}
	return h.reconcileFlannelHostNetworkPolicy(key, node)
```

Reply:
This came from: https://github.com/k3s-io/k3s/blob/master/pkg/node/controller.go#L39-L49

Reply:
Well, that code is wrong too. Event handlers can and will be invoked with a `nil` object.

Reply:
I consolidated into a `handle` function